lib/shoulda/matchers/active_model/validate_confirmation_of_matcher.rb

```diff
@@ -94,28 +94,48 @@ def simple_description
 
         def matches?(subject)
           super(subject)
-
-          disallows_different_value &&
-            allows_same_value &&
-            allows_missing_confirmation
+          type = find_type(subject)
+          disallows_different_value(type) &&
+            allows_same_value(type) &&
+            allows_missing_confirmation(type)
         end
 
         private
 
-        def disallows_different_value
-          disallows_value_of('different value') do |matcher|
-            qualify_matcher(matcher, 'some value')
+        def find_type(subject)
+          subject.class.columns_hash['attribute_to_confirm'].type rescue :string
+        end
+
+        def different_value
```

Comment:
What do you think about defining these as constants rather than methods? Also, what do you think about using the plural (`DIFFERENT_VALUES`) rather than the singular?